tests/apis/anthropic/test_anthropic.py

```diff
@@ -0,0 +1,535 @@
+    "prompt": "test prompt",
+    "parameters": {"temperature": 1, "max_tokens": 100},
+}
+
+PROMPT_DICT_CHAT = {
+    "id": "anthropic_id",
+    "api": "anthropic",
+    "model_name": "anthropic_model_name",
+    "prompt": ["test chat 1", "test chat 2"],
+    "parameters": {"temperature": 1, "max_tokens": 100},
+}
+
+PROMPT_DICT_HISTORY = {
+    "id": "anthropic_id",
+    "api": "anthropic",
+    "model_name": "anthropic_model_name",
+    "prompt": [
+        {"role": "system", "content": "test system prompt"},
+        {"role": "user", "content": "user message"},
+    ],
+    "parameters": {"temperature": 1, "max_tokens": 100},
+}
+
+PROMPT_DICT_HISTORY_NO_SYSTEM = {
+    "id": "anthropic_id",
+    "api": "anthropic",
+    "model_name": "anthropic_model_name",
+    "prompt": [
+        {"role": "user", "content": "user message 1"},
+        {"role": "assistant", "content": "assistant message"},
+        {"role": "user", "content": "user message 2"},
+    ],
+    "parameters": {"temperature": 1, "max_tokens": 100},
+}
+
+TYPE_ERROR_MSG = (
+    "if api == 'anthropic', then the prompt must be a str, list[str], or "
+    "list[dict[str,str]] where the dictionary contains the keys 'role' and "
+    "'content' only, and the values for 'role' must be one of 'user' or 'model', "
+    "except for the first message in the list of dictionaries can be a "
+    "system message with the key 'role' set to 'system'."
+)
+
+
+def test_anthropic_api_init(temporary_data_folders):
+    # raise error if no arguments are provided
+    with pytest.raises(TypeError, match="missing 2 required positional arguments"):
+        AnthropicAPI()
+
+    # raise error if no settings object is provided
+    with pytest.raises(
+        TypeError, match="missing 1 required positional argument: 'settings'"
+    ):
+        AnthropicAPI(log_file="log.txt")
+
+    # raise error if no log file name is provided
+    with pytest.raises(
+        TypeError, match="missing 1 required positional argument: 'log_file'"
+    ):
+        AnthropicAPI(settings=Settings(data_folder="data"))
+
+    settings = Settings(data_folder="data")
+    log_file = "log.txt"
+    anthropic_api = AnthropicAPI(settings=settings, log_file=log_file)
+
+    assert anthropic_api.settings == settings
+    assert anthropic_api.log_file == log_file
+    assert anthropic_api.api_type == "anthropic"
+
+
+def test_anthropic_check_environment_variables(temporary_data_folders, monkeypatch):
+    # only warn on the ANTHROPIC_API_KEY environment variable
+    test_case = AnthropicAPI.check_environment_variables()
+    assert len(test_case) == 1
+    with pytest.raises(
+        Warning, match="Environment variable 'ANTHROPIC_API_KEY' is not set"
+    ):
+        raise test_case[0]
+
+    # no warnings or errors
+    monkeypatch.setenv("ANTHROPIC_API_KEY", "DUMMY")
+    assert AnthropicAPI.check_environment_variables() == []
+
+
+def test_anthropic_check_prompt_dict(temporary_data_folders, monkeypatch):
```

Comment:
Add a docstring or comment summarising what edge cases we're testing here as its quite a long function

Reply:
Actually, each case is individually well commented. ignore this.